ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -692,3 +722,140 @@ func validateTaskNetworkConfig(taskNetworkConfig *state.TaskNetworkConfig) error
 
 	return nil
 }
+
+// checkPacketLossFault checks if there's existing network-packet-loss fault running.
+func (h *FaultHandler) checkPacketLossFault(taskMetadata *state.TaskResponse, request types.NetworkPacketLossRequest) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	interfaceName := taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].NetworkInterfaces[0].DeviceName
+	lossPercent := request.LossPercent
+	networkMode := taskMetadata.TaskNetworkConfig.NetworkMode
+	ipSources := request.Sources
+	// If task's network mode is awsvpc, we need to run nsenter to access the task's network namespace.
+	nsenterPrefix := ""
+	if networkMode == "awsvpc" {
+		nsenterPrefix = fmt.Sprintf(nsenterCommandString, taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].Path)
+	}
+
+	// We will run the following Linux command to assess if there existing fault.
+	// "tc -j q show dev {INTERFACE} parent 1:1"
+	// The command above gives the output of "tc q show dev {INTERFACE} parent 1:1" in json format.
+	// We will then unmarshall the json string and evaluate the fields of it.
+	tcCheckInjectionCommandComposed := nsenterPrefix + fmt.Sprintf(tcCheckInjectionCommandString, interfaceName)
+	fmt.Println(tcCheckInjectionCommandComposed)
+	cmdOutput, err := h.runExecCommand(ctx, tcCheckInjectionCommandComposed)
+	if err != nil {
+		return false, errors.New("failed to check network-packet-loss-fault: " + string(cmdOutput[:]) + err.Error())
+	}
+	// Log the command output to better help us debug.
+	logger.Info(fmt.Sprintf("%s command result: %s", tcCheckInjectionCommandComposed, string(cmdOutput[:])))
+	var outputUnmarshalled []map[string]interface{}
```

Comment:
Q: Instead of a parsing/unmarshaling the output into a `map[string]interface{}`, I wonder if we can parse it into a struct and then get only the needed json keys from the output? (if it's not possible/output is not deterministic then please disregard).

Just ran `tc -j q show dev eth0 parent 1:1` locally and it seems to output the following:
```
[{
    "kind": "netem",
    "handle": "10:",
    "parent": "1:1",
    "options": {
        "limit": 1000,
        "loss-random": {
            "loss": 0.5,
            "correlation": 0
        },
        "ecn": false,
        "gap": 0
    }
}]
```

Reply:
`tc -j q show dev eth0` does not always have all the fields. For example, when running on a host that doesn't have fault injected, it can have the following outout:
```
{"kind":"pfifo_fast","handle":"0:","parent":":1","options":{"bands":3,"priomap":[1,2,2,2,1,2,0,0,1,1,1,1,1,1,1,1],"multiqueue":false}}]
```
There's no loss-random field here. Also, when specifying `parent 1:1`, the output can also be `[]`.

Unmarshalling it into a `map[string]interface{}` and indexing the json struct iteratively would allow us to better catch the `nil` when parsing it